sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -50,60 +188,52 @@ namespace Azure { namespace Core {
       /// RFC 1123.
       Rfc1123,
 
-      /// ISO 8601.
-      Iso8601,
+      /// RFC 3339.
+      Rfc3339,
```

Comment:
Unlike ISO, RFC 3339 doesn't limit the number of fractional second digits. This will need some implementation change (can be done in a separate PR).

I recommend supporting up to 16 fractional digits (ignoring digit 8-16), throwing for anything more than that, but only using the first 7 as meaningful digits.

This RFC also supports ' ', 'T', 'Z', 't', 'z' for separators. What is our behavior for those? It is OK to only accept 'T' and ' ', if you like, and reject the lower case versions.

Look at this for more info on the subtle differences between ISO 8601 and RFC 3339: https://docs.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support#support-for-parsing

Reply:
Yes, that will go as a separate PR, now we want clean API interface.
I also looked at .NET implementation, it does rounding on 8th digit, >=5 is always up. Years 0001 - 9999 are allowed.